test/common/test_helpers.h

```diff
@@ -1,19 +1,91 @@
-// Copyright (C) 2023 Intel Corporation
+// Copyright (C) 2023-2024 Intel Corporation
 // Under the Apache License v2.0 with LLVM Exceptions. See LICENSE.TXT.
 // SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
-// This file contains tests for UMF pool API
+// This file contains helpers for tests for UMF pool API
 
 #ifndef UMF_TEST_HELPERS_H
 #define UMF_TEST_HELPERS_H 1
 
+#include <stdarg.h>
+#include <stdio.h>
 #include <umf/base.h>
 #include <umf/memory_pool.h>
 #include <umf/memory_provider_ops.h>
 
+#ifdef __cplusplus
+#include <stdexcept>
+static inline void UT_FATAL(const char *format, ...) {
```

Comment:
You can just use the C-version of the UT_FATAL (the one under `#else`) since this is only expected to be used by C code anyway.

Reply:
@DamianDuy You are adding asserts for C tests, so you should add only the C version of `UT_FATAL`, not C++

Reply:
Done.